src/lib/secrets.js

```diff
@@ -0,0 +1,149 @@
+const { Buffer } = require('buffer')
```

Comment:
The rest of the package is now written in TypeScript, so it would make sense to do the same here. As you already have the typings in a .d.ts it hopefully wouldn't be too much of a lift.

Reply:
Converting this to TypeScript is proving to be fairly painful. Is there some internal guide for e.g. how to get around the `ts-ignore` ban and whatnot?

Reply:
OK, in the interest of getting this out of the door on a deadline I suggest shipping it as js today, and we can miograte it when we have a bit more time

Reply:
Whoops, missed this, and went through quite a large rewrite - could use another pair of eyes to look it over!